### PR TITLE
Group dependencies for submodules

### PR DESCRIPTION
**Pull request by zpapakipos**
## Related Issue (if applicable)
Fixes https://github.com/facebookresearch/AugLy/issues/65 & https://github.com/facebookresearch/AugLy/issues/75.

## Summary
What are you trying to achieve in this PR? Please summarize what changes you made and how they achieve the desired result.

In order to make AugLy less heavy for users to install, let's separate the heavier dependencies that are specific to the audio/video modules and only require the dependencies for the module that the user specifies. The user can now install augly using one of the following commands, depending which module(s) they want to be able to use:
```bash
pip install augly
pip install augly[av]
```

I moved some of the requirements from the base `requirements.txt` to `av_requirements.txt`, etc for all 4 modules. I also removed any unnecessary requirements & changed to specifying minimum version numbers instead of pinning exact versions.

### Image & Text
```bash
conda create -n augly_group_deps python=3.9 -y && conda activate augly_group_deps
pip install -e . && pip install torchvision

python -m unittest discover -s augly/tests/image_tests/ -p "*"
----------------------------------------------------------------------
Ran 74 tests in 35.072s

OK (skipped=5)

python -m unittest discover -s augly/tests/text_tests/ -p "*"
----------------------------------------------------------------------
Ran 26 tests in 0.138s

OK

conda deactivate && conda env remove -n augly_group_deps -y
```

## Other testing

```bash
conda activate pypi
python -m build --wheel
twine upload dist/*
```
I ran each of the 4 example colab notebooks, installing `augly` or `augly[av]` from the newest version on pypi. For example here is the behavior of the audio notebook now when installing the base package `augly`, which no longer includes the dependencies required for the audio module:
Screenshot 2021-07-08 at 18 56 12

And here is the behavior when installing the package with the av extras, `augly[av]`:
Screenshot 2021-07-08 at 18 59 35

And for the image notebook for example, everything still works when installing `augly` as normal:
Screenshot 2021-07-08 at 17 44 13

I can always yank this latest version of augly off of pypi if we identify an issue during review of this PR. If not, after we land this PR I will make a new github release as well to upgrade the DOI version to align with pypi.

### CODE REVIEW

**Comment by alexsounder** [comment]
@zpapakipos thanks for tackling this so fast. I wanted to mention that dependencies for multiple modalities can be installed with a single command like `augly[audio,text]`. This works already, I just think it should be documented. Also, one observation, there's still a lot of dependencies like `joblib` that are not used directly and in my opinion, should be removed.

File: augly/image/README.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
 # Image
 
+## Installation   
+If you want to use the image augmentations from AugLy, you can install using the following command:
```
Comment (jbitton) [inline augly/image/README.md:4]:
```suggestion
If you would like to use the image augmentations, please install AugLy using the following command:
```

File: augly/text/README.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # Text
 
+## Installation   
+If you plan to use the text augmentations from AugLy, you can install using the following command:
```
Comment (jbitton) [inline augly/text/README.md:4]:
```suggestion
If you would like to use the text augmentations, please install AugLy using the following command:
```

File: augly/video/README.md [hunk 3]
```diff
@@ -1,8 +1,14 @@
 # Video
 
-## Installation Notes
+## Installation
+If you plan to use the video augmentations from AugLy, you can install using the following command:
```
Comment (jbitton) [inline augly/video/README.md:4]:
```suggestion
If you would like to use the video augmentations, please install AugLy using the following command:
```

**Review by jbitton — APPROVED** [review]
Overall LGTM! Please address these small nits before landing!

**Comment by jbitton** [comment]
An additional comment: did you already push this to pip? If so, please make another release of AugLy such that our Zenodo DOI is linked up correctly :)

File: setup.py [hunk 4]
```diff
@@ -11,13 +11,21 @@
...
+}
+
 with open("README.md", encoding="utf8") as f:
     readme = f.read()
 
 
 setuptools.setup(
     name="augly",
-    version="0.1.3",
+    version="0.1.5",
```
Comment (jbitton) [inline setup.py:28]: i think this should be version `0.1.4`, no?

Comment (zpapakipos) [reply setup.py:28]: I already uploaded & then yanked a version 0.1.4 (the previous version of splitting dependencies into extras), so unfortunately that means I can't upload another 0.1.4.

File: README.md [hunk 5]
```diff
@@ -48,12 +48,14 @@ pip install augly
 
 Or clone AugLy if you want to be able to run our unit tests, contribute a pull request, etc:
 ```bash
-git clone git@github.com:facebookresearch/AugLy.git
+git clone git@github.com:facebookresearch/AugLy.git && cd AugLy
 [Optional, but recommended] conda create -n augly && conda activate augly && conda install pip
-pip install -e AugLy/
+pip install -e .
```
[the remark this answers is not in the document]
Comment (zpapakipos) [reply README.md:53]: There is a note about that just below :)

**Pull request merged**